spec/outputs/tcp_spec.rb

```diff
@@ -227,5 +254,59 @@
 
       end
     end
+
+    context "and protocol is TLSv1.3" do
+      let(:key_file) { File.join(FIXTURES_PATH, 'plaintext/instance.key') }
+      let(:crt_file) { File.join(FIXTURES_PATH, 'plaintext/instance.crt') }
+      let(:config) { super().merge("ssl_cert" => crt_file, "ssl_key" => key_file) }
+
+      let(:secure_server) do
+        ssl_context = OpenSSL::SSL::SSLContext.new
+        ssl_context.verify_mode = OpenSSL::SSL::VERIFY_NONE
+        ssl_context.cert = OpenSSL::X509::Certificate.new(File.read(crt_file))
+        ssl_context.key = OpenSSL::PKey::RSA.new(File.read(key_file), nil)
+        ssl_context.min_version = OpenSSL::SSL::TLS1_3_VERSION
+        OpenSSL::SSL::SSLServer.new(server, ssl_context)
+      end
+
+      before(:each) {
+        subject.register
+      }
+
+      after(:each) {
+        secure_server.close rescue nil
+      }
+
+
+      # This test confirms that this plugin is able to write to a TLS socket
+      # multiple times.
+      # Previous implementation performed an IO.select first and called sysread
+      # if select signaled the socket was ready to read.
+      # For TLS1_3, due to control messages it may happen that the underlying
+      # socket is marked as readable but there is no new data available,
+      # causing a read to block forever.
+      # This test will raise a Timeout exception with the old implementation.
+      it 'successfully writes two messages' do
+        message = "a"
+
+        buffer = ""
```

Comment:
```suggestion
      let(:message) { "a" }
      let(:buffer) { "" }
        
      # This test confirms that this plugin is able to write to a TLS socket
      # multiple times.
      # Previous implementation performed an IO.select first and called sysread
      # if select signaled the socket was ready to read.
      # For TLS1_3, due to control messages it may happen that the underlying
      # socket is marked as readable but there is no new data available,
      # causing a read to block forever.
      # This test will raise a Timeout exception with the old implementation.
      it 'successfully writes two messages' do

```

Reply:
For the sake of learning, why should we prefer to declare those variables using `let`, if they won't be used by any other test case in this context?

Reply:
I've applied your suggestion. Thanks for reviewing! 👍